src/DotPulsar/PulsarClient.cs

```diff
@@ -118,12 +119,58 @@ public async Task<PartitionedTopicMetadata> GetPartitionTopicMetadata(string top
         public IConsumer CreateConsumer(ConsumerOptions options)
         {
             ThrowIfDisposed();
+
+            PartitionedTopicMetadata partitionedTopicMetadata = GetPartitionTopicMetadata(options.Topic).Result;
```

Comment:
Regarding the earlier discussion about using `.GetAwaiter().GetResult()`, this will throw an AggregateException containing the "real" exception.  Is there a reason to prefer that, here?